dbms/src/DataStreams/SummingSortedBlockInputStream.h

```diff
@@ -69,15 +69,20 @@ class SummingSortedBlockInputStream : public MergingSortedBlockInputStream
      *   and can be deleted at any time.
      */
 
-    /// Stores numbers of key-columns and value-columns.
-    struct MapDescription
+    struct AggregateDescription
     {
+        AggregateFunctionPtr function;
+        std::vector<size_t> column_numbers;
+        ColumnPtr merged_column;
+        std::vector<char> state;
+        bool created = false;
+        /* Compatibility with the mergeMap */
```

Comment:
Why not separate struct?

Reply:
Because I don't know whether the legacy `mergeMap`, or `sumMap` will be used when identifying maps in block, so instead of building two structures, I'm using just one. I'm happy to rework it to separate structs, it doesn't matter much.